src/wp-includes/class-wp-user-query.php

```diff
@@ -790,6 +793,11 @@ public function query() {
 
 		$qv =& $this->query_vars;
 
+		// Do not cache results if more than 3 fields are requested.
+		if ( is_array( $qv['fields'] ) && count( $qv['fields'] ) > 3 ) {
```

Comment:
@peterwilsoncc I have defined this as 3. But do you have thoughts this value. 3 is just a number I picked out of the air.

Reply:
> 3 is just a number I picked out of the air.

Three works for me. In my half thoughts, I'd picked the same number out of the air too :)